src/main/scala/strawman/collection/IterableFactory.scala

```diff
@@ -0,0 +1,35 @@
+package strawman
+package collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Int
+
+/** Base trait for instances that can construct a collection from an iterable */
+trait FromIterable[+C[X] <: Iterable[X]] {
+  def fromIterable[B](it: Iterable[B]): C[B]
+}
+
+/** Base trait for builder factories */
+trait CanBuild[-A, +Repr] {
```

Comment:
Actually, I’m thinking that `CanBuild` also has the following advantages:
- Simpler to use for end-users (e.g. the type of the parameter is `CanBuild[A, C[A]]` instead of `() => Builder[A, C[A]]`) ;
- We could add other methods than just `apply(): Builder[A, C[A]]`, like `apply(knownSize: Int): Builder[A, C[A]]`.